src/parsers/expression/selection.js

```diff
@@ -93,8 +96,7 @@ function vlSelection(name, unit, datum, op, scope, test) {
 
   // if intersecting and we made it here, then we saw no misses
   // if not intersecting, then we saw no matches
-  // if no active selections, return true
-  return !n || intersect;
+  return intersect;
```

Comment:
Could remove check at L79 and replace return line here with: `return n && intersect;`